examples/dperf/dperf.c

```diff
@@ -15,129 +15,228 @@
  * along with Packetgraph.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdlib.h>
+#include <stdio.h>
 #include <unistd.h>
 #include <signal.h>
 #include <glib.h>
 #include <errno.h>
 #include <arpa/inet.h>
 #include <sys/time.h>
 #include <time.h>
-
-#include <rte_config.h>
-#include <rte_ether.h>
-#include <rte_ethdev.h>
-#include <rte_ip.h>
-
+#include <string.h>
+#include <netinet/ether.h>
+#include <linux/if_ether.h>
+#include <linux/ip.h>
+#include <linux/udp.h>
+#include <arpa/inet.h>
 #include <packetgraph/packetgraph.h>
-#include <packetgraph/nic.h>
-#include <packetgraph/vhost.h>
-#include <packetgraph/print.h>
-#include <packetgraph/brick.h>
-#include <packetgraph/utils/mempool.h>
-#include <packetgraph/utils/bitmask.h>
-#include <packetgraph/utils/mac.h>
-#include <packetgraph/packets.h>
 
 #define CHECK_ERROR(error) do {			\
 		if (error)			\
 			pg_error_print(error);	\
 		g_assert(!error);		\
 	} while (0)
 
-static int start_loop(int32_t ip_src, int32_t ip_dst,
-		      struct ether_addr src_mac,
-		      struct ether_addr dst_mac,
-		      int mtu)
-{
-	int ret = -1;
-	uint64_t mask = ~0LL;
-	struct rte_mbuf **pkts = pg_packets_create(mask);
-	struct pg_error *error = NULL;
-	struct pg_brick *nic = pg_nic_new_by_id("nic", 1, 1,
-						WEST_SIDE, 0, &error);
-	CHECK_ERROR(error);
-	char	pkt[1500];
+#define DATA_LEN 1000
 
-	memset(pkt, ~0, 1500);
-	pkt[1499] = '\0';
+struct packet {
+	struct ethhdr eth;
+	struct iphdr ip;
+	struct udphdr udp;
+	char payload[DATA_LEN];
+} __attribute__((__packed__));
 
-	pg_packets_append_ether(pkts, mask, &src_mac,
-				&dst_mac, ETHER_TYPE_IPv4);
-	pg_packets_append_ipv4(pkts, mask, ip_src, ip_dst, mtu, 253);
-	pkt[mtu - sizeof(struct ipv4_hdr) - sizeof(struct ether_hdr)] = '\0';
-	pg_packets_append_str(pkts, mask, pkt);
+#define PKT_LEN sizeof(struct packet)
 
-	while (1) {
-		pg_brick_burst_to_west(nic, 0, pkts, 64, mask, &error);
-		CHECK_ERROR(error);
-		usleep(1);
+static void tx_callback(struct pg_brick *brick,
+			pg_packet_t **tx_burst,
+			uint16_t *tx_burst_len,
+			void *private_data)
+{
+	static bool copied = false;
```

Comment:
I think linux style is more to let copied uninitialized, as since it's static it will be initialize to zero anyway.